misc/python/materialize/cli/mzcompose.py

```diff
@@ -24,43 +24,105 @@
 MIN_COMPOSE_VERSION = (1, 24, 0)
 
 
-def main(argv: List[str]) -> int:
-    # Lightly parse the arguments so we know what to do.
-    parser = ArgumentParser()
-    args, unknown_args = parser.parse_known_args(argv)
-    if args.help or args.command == "help":
-        parser.print_help()
-        return 0
+def main(argv: List[str]) -> None:
+    parser = ArgumentParser(prog="mzcompose")
 
+    # Global arguments. For all arguments but `--mz-quiet` the `--mz` prefix
+    # is accepted for backwards compatibility.
+    parser.add_argument(
+        "--mz-quiet",
+        action="store_true",
+        help="suppress Materialize-specific informational messages",
+    )
+    parser.add_argument(
+        "--find",
+        "--mz-find",
+        metavar="DIR",
+        help="use the mzcompose.yml file from DIR, rather than the current directory",
+    )
+    parser.add_argument(
+        "--build-mode",
+        "--mz-build-mode",
+        default="release",
+        choices=["dev", "release"],
+        help="specify the Cargo profile to use when compiling Rust crates",
+    )
+    parser.add_argument(
+        "--coverage",
+        "--mz-coverage",
+        action="store_true",
+        help='emit code coverage reports to the "coverage" directory',
+    )
+
+    # Docker Compose arguments that we explicitly ban. Since we don't support
+    # these, we hide them from the help output.
+    parser.add_argument("-f", "--file", nargs="?", help=argparse.SUPPRESS)
+    parser.add_argument("--project-directory", nargs="?", help=argparse.SUPPRESS)
```

Comment:
if they pass these, do we want a runtime exception?

Reply:
Yes! That happens on L99 (and matches what currently happens).

Reply:
ah I missed that!